src/DSP.jl

```diff
@@ -39,24 +44,65 @@ for (T, suff) in ((Float64, "D"), (Float32, ""))
 end
 
 
-function plan_dct(n,k::Integer)
-    @assert isinteger(Base.log2(n))
-    @assert 2≤k≤4
-    ccall(("vDSP_DCT_CreateSetup",libacc),Ptr{Void},(Ptr{Void},Cint,Cint),C_NULL,n,k)
+## == Discrete Cosine Transform (DCT) == ##
+"""
+Initializes a new DCT setup object. 'dct_type' must be 2, 3, 4 corresponding to Type II, III and IV.
+DCT 'length' must be equal to f*(2^n) where f = 1,3,5,15 and n >= 4. If you have a previous DCT setup
+object, that can be passed in as 'previous'. The returned DCT setup will share the underlying data
+storage of the previous setup object.
+
+Returns: DFTSetup
+"""
+function plan_dct(length::Integer,  dct_type::Integer, previous=C_NULL)
+    n = trailing_zeros(length)
+    f = length >> n
+    if dct_type < 2 &&  dct_type > 4
+        error("DCT type ", dct_type, " is not supported. Only DCT types 2, 3 and 4 are supported")
+    elseif !(n >= 4 && f in (1,3,5,15))
+        error("Invalid DCT length. Length must be equal to f*(2^n) where f = 1,3,5,15 and n >= 4")
+    end
+    setup::Ptr{Void} = ccall(("vDSP_DCT_CreateSetup", libacc), Ptr{Void},
+                             (Ptr{Void}, UInt64, UInt64),
+                             previous, length, dct_type)
+    return DFTSetup{Float32}(setup, 0)
```

Comment:
Standard practice is to attach a finalizer, so this gets cleaned up when the object is GC'ed. You may need to change the `DFTSetup` to a mutable type for this though.